ignite/base/mixins.py

```diff
@@ -22,3 +23,52 @@ def load_state_dict(self, state_dict: Mapping) -> None:
         opts = [k in state_dict for k in self._state_dict_one_of_opt_keys]
         if len(opts) > 0 and ((not any(opts)) or (all(opts))):
             raise ValueError("state_dict should contain only one of '{}' keys".format(self._state_dict_one_of_opt_keys))
+
+
+class Attachable:
+    """
+    This class contains an internal dictionary `attach_map` associating events with the bound methods plus args/kwargs
+    to be triggered on those events. This dictionary can be populated by the contructor or by external clients after
+    construction and before `attach` is called.
+    """
+
+    def __init__(self, attach_map={}):
+        self.attach_map = dict(attach_map)
+
+    def _iter_handlers(self):
+        for event, (method, _, _) in self.attach_map.items():
+            yield event, method
+
+    def set_handler_mapping(self, event_name: Any, handler: Callable, *args, **kwargs) -> None:
+        if handler is not None:
+            self.attach_map[event_name] = (handler, args, kwargs)
+        else:
+            del self.attach_map[event_name]
+
+    def attach(self, engine: Any) -> None:
```

Comment:
The argument engine should be typed as Engine, shouldn’t it?

Reply:
The problem is that I would have to import `Engine` from its module, since that module imports this one this creates a cyclic dependency between them. `Engine` could be imported within the method and type check done there.